tape/src/test/java/com/squareup/tape/FileObjectQueueTest.java

```diff
@@ -4,31 +4,131 @@
 import java.io.File;
 import java.io.IOException;
 import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
+import static com.squareup.tape.QueueElementObserver.NO_OP_OBSERVER;
+import static com.squareup.tape.QueueFile.MAX_ELEMENT_SIZE;
+import static com.squareup.tape.QueueTestUtils.ONE_ENTRY_SERIALIZED_QUEUE;
+import static com.squareup.tape.QueueTestUtils.copyTestFile;
 import static org.fest.assertions.Assertions.assertThat;
 
 public class FileObjectQueueTest {
+  private static final long TIMEOUT_MS = 200;
   @Rule public TemporaryFolder folder = new TemporaryFolder();
+  private File parent;
   private FileObjectQueue<String> queue;
 
   @Before public void setUp() throws IOException {
-    File parent = folder.getRoot();
+    parent = folder.getRoot();
     File file = new File(parent, "queue-file");
     queue = new FileObjectQueue<String>(file, new SerializedConverter<String>());
     queue.add("one");
     queue.add("two");
     queue.add("three");
   }
 
+  @After public void cleanUp() {
+    queue.close();
+  }
+
   @Test public void peekMultiple() {
     List<String> peek = queue.peek(3);
     assertThat(peek).containsExactly("one", "two", "three");
   }
 
+  @Test public void allElementOperationsCall() throws IOException, InterruptedException {
+    File file = new File(parent, "queue-file-alpha");
+    CountDownLatch latch = new CountDownLatch(6);
+    queue = new FileObjectQueue<String>(file,
+        new SerializedConverter<String>(),
+        MAX_ELEMENT_SIZE,
+        (observation) -> latch.countDown());
+    queue.add("one");
+    queue.add("two");
+    queue.add("three");
+    List<String> peek = queue.peek(3);
+    assertThat(peek).containsExactly("one", "two", "three");
+    latch.await(TIMEOUT_MS, TimeUnit.MILLISECONDS);
```

Comment:
Don't use a latch. There is nothing async here. Plus this returns `false` if the timeout is triggered so you're missing an `assertTrue` on each one.

Use an `AtomicBoolean`, `set(true)`, and `assertTrue(whevs.get())`.